wallet/src/ui/app/redux/slices/sui-objects/Coin.ts

```diff
@@ -26,4 +32,28 @@ export class Coin {
     public static getBalance(obj: SuiMoveObject) {
         return BigInt(obj.fields.balance);
     }
+
+    /**
+     * Transfer `amount` of Coin<T> to `recipient`.
+     *
+     * @param signer A signer with connection to the gateway:e.g., new RawSigner(keypair, new JsonRpcProvider(endpoint))
+     * @param coins A list of Coins owned by the signer with the same generic type(e.g., 0x2::Sui::Sui)
+     * @param amount The amount to be transfer
+     * @param recipient The sui address of the recipient
+     */
+    public static async transferCoin(
+        signer: RawSigner,
```

Comment:
cc @pchrysochoidis I will iterate on the implementation of this function, but let me know if the function signature makes sense